### PR TITLE
add experimental support for kube-prometheus stack

### PR DESCRIPTION
**Pull request by VineethReddy02**
Below cmd will install Kube-Prometheus in tobs and will disable default Prometheus & Grafana deployments as Kube-Prometheus will deploy those components. 

```
tobs install -c ./chart/ --enable-kube-prometheus
```

**Note**: This PR is dependent on #114 as there is some common functionality & common refactoring. This PR is rebased on top of that PR. 

Todos:

1. Documents the Kube-Prometheus add-on. 
2. Add new tobs CLI commands to manage & configure Kube-Prometheus.
3. Add an option to deploy Kube-Prometheus with Prometheus HA.

### CODE REVIEW

File: chart/values.yaml [hunk 1]
```diff
@@ -159,6 +159,22 @@ promlens:
...
 
+kube-prometheus-stack:
+  enabled: false
+  prometheus:
+    prometheusSpec:
+      ## The remote_read spec configuration for Prometheus.
+      ## ref: https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#remotereadspec
+      remoteRead:
+      # - {protocol}://{host}:{port}/{endpoint}
+        - url: "http://tobs-promscale-connector.default.svc.cluster.local:9201/read"
```
Comment (blagojts) [inline chart/values.yaml:170]: can we use the same template that the prom part of the chart uses?

File: cli/cmd/helmInstall.go [hunk 2]
```diff
@@ -50,14 +51,19 @@ func helmInstall(cmd *cobra.Command, args []string) error {
...
+
+	err = installStack(file, ref, dbURI, kp)
 	if err != nil {
 		return fmt.Errorf("could not install The Observability Stack: %w", err)
 	}
 	return nil
 }
 
-func installStack(file, ref, dbURI string) error {
+func installStack(file, ref, dbURI string, kp bool) error {
```
Comment (blagojts) [inline cli/cmd/helmInstall.go:66]:
```suggestion
func installStack(file, ref, dbURI string, enableKubePrometheus bool) error {
```

File: cli/cmd/helmInstall.go [hunk 3]
```diff
@@ -119,11 +132,13 @@ func installStack(file, ref, dbURI string) error {
...
 
-func appendDBURIValues(dbURI, name string, cmds []string) ([]string, error) {
-	cmds = append(cmds, "--set",
-		"timescaledb-single.enabled=false,"+
-			"timescaledbExternal.enabled=true,"+
-			"timescaledbExternal.db_uri="+dbURI+
-			",promscale.connection.uri.secretTemplate="+name+"-timescaledb-uri,")
-	return cmds, nil
+func appendDBURIValues(dbURI, name string, helmValues string) string {
+	helmValues = helmValues+",timescaledb-single.enabled=false,"+"timescaledbExternal.enabled=true,"+"timescaledbExternal.db_uri="+dbURI+
```
Comment (blagojts) [inline cli/cmd/helmInstall.go:136]: tl;dr

**Review by blagojts — CHANGES_REQUESTED** [review]

File: chart/values.yaml [hunk 4]
```diff
@@ -159,6 +159,22 @@ promlens:
   loadBalancer:
     enabled: false
```
Comment (cevian) [inline chart/values.yaml:161]: We should put some comments about this being experimental and also highlight which components should be disabled if this is enabled. In fact, I might have a kube_prometheus_values.yml file in this dir that would serve as an example of what that would look like

**Review by cevian — CHANGES_REQUESTED** [review]
Overall looks good, but I think we need much better documentation

**Review by cevian — COMMENTED** [review]
should we have tobs uninstall kube-prometheus crds automatically https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack#uninstall-chart

File: cli/cmd/helmInstall.go (hunk 3 above)
Comment (VineethReddy02) [reply cli/cmd/helmInstall.go:136]: This refactoring of existing logic. When connecting to external DB. Tobs needs to disabled TimescaleDB installation and map values to timescaleDBexternal fields accordingly.

**Pull request closed without merging**